src/RegExpr.pas

```diff
@@ -4247,20 +4242,26 @@ function TRegExpr.MatchAtOnePos(APos: PRegExprChar): boolean;
 end;
 
 procedure TRegExpr.ClearMatches;
+begin
+  FillChar(startp, SizeOf(startp), 0);
+  FillChar(endp, SizeOf(endp), 0);
+end;
+
+procedure TRegExpr.ClearInternalIndexes;
 var
   i: integer;
 begin
+  FillChar(startp, SizeOf(startp), 0);
+  FillChar(endp, SizeOf(endp), 0);
   for i := 0 to NSUBEXP - 1 do
-  begin
-    startp[i] := nil;
-    endp[i] := nil;
-  end;
+    GrpIndexes[i] := -1;
+  GrpIndexes[0] := 0;
+  GrpCount := 0;
 end;
 
-function TRegExpr.ExecPrim(AOffset: integer; ATryOnce: boolean): boolean;
+function TRegExpr.ExecPrim(AOffset: integer; ATryOnce, ASlowChecks: boolean): boolean;
```

Comment:
Looks like you just switch off optimization completely
You use ASlowChecks = False and do not use regmustString in this case.
BTW this is contr-intuitive - we slow check if ASlowCheck is False.